packages/zoltan2/test/helpers/AdapterForTests.hpp

```diff
@@ -92,25 +92,21 @@ using std::string;
 using namespace Zoltan2_TestingFramework;
 
 // helper struct to store both an adapter and the coordinate adapter
-struct AdapterWithOptionalCoordinateAdapter
+struct AdapterWithTemplateName
 {
-  Zoltan2_TestingFramework::base_adapter_t * mainAdapter = nullptr;
-  Zoltan2::VectorAdapter<tMVector_t> * coordinateAdapter = nullptr;  // may be NULL if coordinates are not available
+  Zoltan2::BaseAdapterRoot * adapter = nullptr; // generic base class
```

Comment:
Correct the testing code is storing the BaseAdapterRoot as a ptr with no adapter reference. I believe I have eliminated all casts to basic_id_t. If there are any left it is an error as I intended to eliminate them completely. In the new form there should never be any unsafe conversions and no possibility of accidentally working with an object casted to the wrong type.